docs/content/develop/step-2e.md

```diff
@@ -0,0 +1,319 @@
+          "type": "string",
+          "default": "pfe-cool-element:follow"
+        }
+      }
+    }
+  },
+  "required": ["slots", "attributes", "events"],
+  "additionalProperties": false
+}
+```
+
+## General information
+
+At the top of our schema is general information about our component: title, description, the tag name and class, and the category of component it falls under.  We also see the version of the schema draft we are using here.
+
+```
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Cool element",
+  "description": "This element has a profile picture and a follow button.",
+  "type": "object",
+  "tag": "pfe-cool-element",
+  "class": "PfeCoolElement",
+  "category": "content",
+```
+
+A schema is a JSON object with certain protected keys: `title`, `description`, `type`, and `required` are top among these.  For our web component, we specify the slots, attributes, and events as nested objects in the `properties` object of the overall component.
+
+## Slots 
+
+We list out our slots in the properties object of the schema.  Every slot is an array type and can contain either specific tags or a generic `raw` reference which means any HTML can be accepted here.  If the slot in question is a named slot such as `<slot name="pfe-cool-element--region">`, change the `"namedSlot"` boolean to true.
+
+```
+"slots": {
+  "title": "Slots",
+  "description": "Definition of the supported slots",
+  "type": "object",
+  "properties": {
+    "default": {
+      "title": "Default",
+      "type": "array",
+      "namedSlot": false,
+      "items": {
+        "oneOf": [
+          {
+            "$ref": "raw"
+          }
+        ]
+      }
+    }
+  }
+}
+```
+
+## Attributes
+
+Our component supports two attributes: `pfe-following` and `pfe-photo-url`. Both attributes are prefixed with `pfe-` to protect their namespacing and prevent collisions with protected attributes in HTML.  To indicate this prefixing, set `"prefixed": true` on the object for that attribute.  Some attribute names will not be prefixed; in those cases, set the prefixed key to false.  Attributes are listed as properties and can use a variety of types.  Most will probably be of type `string` or `boolean` but `number` is also supported.
+
+```
+"attributes": {
+  "title": "Attributes",
+  "type": "object",
+  "properties": {
+    "following": {
+      "title": "Follow",
+      "type": "boolean",
+      "default": false,
+      "prefixed": true,
+      "observer": "_followToggle"
+    },
+    "photo-url": {
+      "title": "Profile image",
+      "type": "string",
+      "observer": "_addImage"
+    }
+  },
+  "required": ["following]
+}
+```
+
+Note the required array at the bottom of the attributes property object: `"required": ["following]`.  To make an attribute required, meaning it will inject the attribute with it's default value if none is provided, add the key name to the required array.  Multiple values are comma separated.
```

Comment:
```suggestion
Note the required array at the bottom of the attributes property object: `"required": ["following]`.  This will set the `pfe-following` attribute as required. This means that if the attribute is not manually added to the component tag in the source, i.e.  `<pfe-cool-element pfe-following>`, then this attribute will be automatically added with a default value (if provided in the schema) when the component upgrades.  
```